### PR TITLE
Lgg/feat/calculate entries

### LINKED ISSUE #27: Add info about last period and length to user profile
- Information we need: When was the users last flow (Make flow not nullable in form) 
- How many days did the flow last? 
- Make an endpoint to get this information and give it to Amber for Open AI API request 
-

### PR DESCRIPTION
**Pull request by lauraguerra1**
## What I did:
closes #27 
- Refactored the add entry endpoint to no longer require a large conditional or inserting or updating
- Added a query to the add entry endpoint to update the last cycle start date and length on a user's profile
- Added the information about a uses last cycle to their profile
- Added an empty state in case a user deletes the entries for their only cycle

## Testing Notes: 
- try changing dates on entries / making new ones and see that data change on user profile page 
## Did this break anything?
- [x]  No
- [ ]  Yes
- [ ]  Potentially (explain)

## Type of change
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

## Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  The code is DRY. If it's not, I tried my best (heavy on the tried my best..)
- [x]  I reviewed my code before pushing

## Screenshots of UI
Screen Shot 2023-10-22 at 8 36 21 PM
Screen Shot 2023-10-22 at 8 40 02 PM

## What's next?
connect to open ai

### CODE REVIEW

File: src/pages/api/addEntry.ts [hunk 1]
```diff
@@ -15,59 +15,81 @@ export default async function addEntry(
...
-      
-        if (error) {
-          return res.status(400).json(error);
+    if (error) {
+      return res.status(400).json(error);
+    }
+    
+    const command = !entries?.length ? 'insert' : 'update'
+    const firstEq = !entries?.length ? 'user_id' : 'date'
+    const secondEq = !entries?.length ? user_id : date
```
Comment (espressoGoddess) [inline src/pages/api/addEntry.ts:30]: wow this is a clever way to do this!

**Review by espressoGoddess — APPROVED** [review]
This looks great! You creatively solved some problems (like whether to insert or update entries) and I also like how you change the profile page to show the last period date without us having to use passage for that

File: src/pages/api/addEntry.ts (hunk 1 above)
Comment (lauraguerra1) [reply src/pages/api/addEntry.ts:30]: Thanks Amber!

**Review by CaliHam — APPROVED** [review]
I'm a little confused about the `last_cycle_start` and `last_cycle_length` and how they are getting updated initially in supabase. Is that something I should add in my code after the registration form is filled out? Or will that be taken care of here?

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -13,15 +13,15 @@ const Navbar = () => {
         <nav className='flex fixed bottom-0 h-20 w-full bg-lightgray'>
             <div className='flex w-full h-20 icon-container items-center'>
                 <div className='h-5/6 w-20 flex p-5 justify-center flex-col items-center'>
-                    <Link href={`${router.asPath.includes('demo') ? '/demo' : ''}/calendar`}>
+                    <Link className='flex flex-col items-center jsutify-center' href={`${router.asPath.includes('demo') ? '/demo' : ''}/calendar`}>
```
[the remark this answers is not in the document]
Comment (lauraguerra1) [reply src/components/Navbar.tsx:16]: lolllll! yeah it probably isn't doing anything lmao. thanks for catching that

[the remark this answers is not in the document]
Comment (lauraguerra1) [reply src/components/Navbar.tsx:16]: @CaliHam You won't have to worry about adding it in your code! The endpoint you are hitting will take care of it for you!

[the remark this answers is not in the document]
Comment (CaliHam) [reply src/components/Navbar.tsx:16]: Okay sweet! Thank you!!

**Pull request merged**